web/workflow.R

```diff
@@ -39,29 +39,36 @@ if (is.na(args[1])){
   settings <- read.settings(settings.file)
 }
 
+# Update/fix/check settings. Will only run the first time it's called, unless force=TRUE
+settings <- fix.deprecated.settings(settings, force=FALSE)
+settings <- addSecrets(settings, force=FALSE)
+settings <- update.settings(settings, force=FALSE)
+settings <- check.settings(settings, force=FALSE)
+
 # Check for additional modules that will require adding settings
 if("benchmark" %in% names(settings)){
   library(PEcAn.benchmark)
   settings <- papply(settings, read.settings.RR)
 }
 
 # Write pecan.CHECKED.xml
-settings <- write.settings(settings, outputfile = "pecan.CHECKED.xml")
+write.settings(settings, outputfile = "pecan.CHECKED.xml")
```

Comment:
@rykelly not sure I follow what's going on here. My understanding was that @bcow deliberately split read.settings into two parts because she needed to put read.settings.RR in BEFORE settings were cleaned up so that she could leverage the checking/updating of settings and the filling in of machine-specific paths. You've now moved that ahead, which will break Benchmarking. 

@bcow also put all those misc helper functions inside of write.settings function so that it was easier to maintain. As per discussions on previous PR #1095, write.settings was a bad choice of name for that function because it is much more intuitive that write.settings would ONLY write out the settings (which I think is how you're using it), but I'd prefer it if there was a simple named single function here [something].settings that retains the functionality of the previous write.settings @bcow wrote